s3transfer/utils.py

```diff
@@ -296,7 +298,8 @@ def __init__(self, filename, start_byte=0, mode='rb', open_function=open):
     def _open_if_needed(self):
         if self._fileobj is None:
             self._fileobj = self._open_function(self._filename, self._mode)
-            self._fileobj.seek(self._start_byte)
+            if self.seekable:
```

Comment:
Ahh I see why you added this now. I wonder if we can get away with something simpler like add a parameter to the ``__init__`` to ensure we do not seek to start or maybe even reusing ``start_byte`` to have the meaning of ``None`` be to not seek anywhere to start?